extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.js

```diff
@@ -0,0 +1,426 @@
+					}
+				}
+
+				function getAvatars(actors) {
+					return actors.slice(0, 5).map(function (avatar) {
+						if (!avatar.avatarUrl) {
+							avatar.avatarUrl = avatarPlaceholder;
+						}
+						avatar.profileUrl = '/wiki/User:' + avatar.name;
+						return avatar;
+					});
+				}
+
+				return notifications.map(this.proxy(function (notification) {
+					return {
+						icon: getIcon(notification.type),
+						uri: notification.uri,
+						showSnippet: !notification.title,
+						snippet: notification.snippet,
+						text: this.textFormatter.getText(notification),
+						isUnread: notification.isUnread,
+						communityName: notification.communityName,
+						showAvatars: notification.totalUniqueActors > 2
+						&& notification.type === common.notificationTypes.discussionReply,
+						showAvatarOverflow: notification.totalUniqueActors > 5,
+						avatarOverflow: notification.totalUniqueActors - 5,
+						avatars: getAvatars(notification.latestActors),
+						timeAgo: $.timeago(notification.when)
+					}
+				}));
+			};
+
+			this.renderNotifications = function (notifications) {
+				templating.renderByLocation(template, this._mapToView(notifications))
+					.then(this.proxy(function (html) {
+						this.$container.append(html);
+						this._bindMarkAsReadHandlers();
+					}));
+			};
+
+			this._bindMarkAsReadHandlers = function () {
+				$(this.$container).find('.wds-notification-card__icon-wrapper')
+					.click(this.proxy(this._markAsRead));
+			};
+
+			this._markAsRead = function (e) {
+				try {
+					var id = $(e.target).closest('.wds-notification-card').attr('data-uri');
+					this.logic.markAsRead(id);
+				} catch (e) {
+					log('Failed to mark as read ' + e, log.levels.error, common.logTag);
+				}
+				return false;
+			};
+
+			this.renderZeroState = function () {
+				$('.wds-notifications__zero-state').addClass(isVisibleClass);
+			};
+
+			this.renderUnreadCount = function (count) {
+				this.unreadCount = count;
+
+				if (this.unreadCount > 0) {
+					this.$markAllAsReadButton.addClass(isVisibleClass);
+					this.$notificationsCount.html(this.unreadCount).parent('.bubbles').addClass('show');
+				} else {
+					this.$markAllAsReadButton.removeClass(isVisibleClass);
+					this.$notificationsCount.empty().parent('.bubbles').removeClass('show');
+				}
+			};
+
+			/**
+			 * .classRemove does not work on SVG
+			 * @param element
+			 */
+			function removeIsUnreadClass(element) {
+				element.classList.remove('wds-is-unread');
+			}
+
+			function findUnreadAndClearClass($element) {
+				$element.find('.wds-icon.wds-is-unread').each(function (_, e) {
+					removeIsUnreadClass(e);
+				});
+			}
+
+			this.renderAllNotificationsAsRead = function () {
+				findUnreadAndClearClass(this.$container);
+			};
+
+			this.renderNotificationAsRead = function (id) {
+				var container = this.$container.find('[data-uri="' + id + '"]');
+				findUnreadAndClearClass(container);
+			};
+
+			this.proxy = function (func) {
```

Comment:
I think I already commented on this method: `$.proxy()` is passé, we try to use `.bind()` in modern js applications.